python/mxnet/model.py

```diff
@@ -69,15 +69,14 @@ def _create_sparse_kvstore(kvstore):
     update_on_kvstore : bool. Always True.
     """
     # always update on kvstore
-    update_on_kvstore = True
     if isinstance(kvstore, kvs.KVStore):
         kv = kvstore
     elif isinstance(kvstore, str):
         kv = kvs.create(kvstore)
     else:
         raise TypeError("Cannot create '%s' KVStore with row_sparse parameters. "
                         "The type must be KVStore or str." % kvstore)
-    return (kv, update_on_kvstore)
```

Comment:
row_sparse weight requires `update_on_kvstore=True`.  The weights are stored in the server, and the worker only pull a subset of the weight in each iteration for mini-batch training. 
Could you instead add an assertion here `assert kv.is_capable(kvs.KVStoreBase.OPTIMIZER) is True`?